### PR TITLE
fix #18674 --nimcache now works better with --os:windows

### LINKED ISSUE #18674: --nimcache and --os:windows interact badly
## example 1
if nimcache is set in a nims config file, eg via:
```nim
import os
let dir = "/tmp/D20210811T023837" / "nimcache1"
# echo dir: \tmp\D20210811T023837\nimcache1
switch("nimcache", dir)
```
it will interact badly with --os:windows, generating the cache in $PWD/\tmp\D20210811T023837\nimcache1

and producing annoying files in $PWD that can't be removed without special escaping syntax, instead of generating those in /tmp/D20210811T023837/nimcache1; eg:
```
git status
 "\\tmp\\D20210811T023837\\nimcache1/"
```
## example 2
ditto with relative paths, eg `let dir = "D20210811T023837" / "nimcache1"`

### Additional Information
1.5.1 6b3c77e7f4421267c551b1cd84717a19c65be2c5

### PR DESCRIPTION
**Pull request by timotheecour**
fix https://github.com/nim-lang/Nim/issues/18674

### CODE REVIEW

File: compiler/commands.nim [hunk 1]
```diff
@@ -537,6 +537,10 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
     conf.lazyPaths.keepItIf(it != path)
   of "nimcache":
     expectArg(conf, switch, arg, pass, info)
+    var arg = arg
+    # otherwise `--os:windows` messes up with `--nimcache` set in config nims files, e.g. via:
+    # import os; switch("nimcache", "/tmp/somedir")
+    if conf.target.targetOS == osWindows and DirSep == '/' and arg.startsWith '\\': arg = arg.replace('\\', '/')
```
Comment (Araq) [inline compiler/commands.nim]: But why would it start with an "\\"?

Comment (timotheecour) [reply compiler/commands.nim]: good point, see last commit which makes this work more robustly, and makes it work for Example 2 i just added in https://github.com/nim-lang/Nim/issues/18674

**Comment by timotheecour** [comment]
PTAL

**Comment by Araq** [comment]
Still an ad-hoc fix, who knows what other switches are affected.

**Comment by timotheecour** [comment]
> Still an ad-hoc fix, who knows what other switches are affected.

possibly other switches need similar fixes, but these can be fixed in future work; ability to run `--os:windows` with wine is a huge time saver when trying to test something on windows from non-windows OS compared to having to spin up a VM / parallels etc, and the 3 other recent PRs improves this workflow

**Pull request merged**